src/node.cc

```diff
@@ -1100,18 +1100,76 @@ int Start(int argc, char** argv) {
     return result.exit_code;
   }
 
-  {
+  if (!per_process::cli_options->snapshot_main.empty()) {
+    SnapshotData data;
+    {
+      std::string entry;
+      int r =
+          ReadFileSync(&entry, per_process::cli_options->snapshot_main.c_str());
+      if (r != 0) {
+        const char* code = uv_err_name(r);
+        const char* message = uv_strerror(r);
+        FPrintF(stderr,
+                "Failed to open %s. %s: %s\n",
+                per_process::cli_options->snapshot_main.c_str(),
+                code,
+                message);
+        return 1;
+      }
+      node::SnapshotBuilder::Generate(
+          &data, entry, result.args, result.exec_args);
+    }
+
+    std::string snapshot_blob_path;
+    if (!per_process::cli_options->snapshot_blob.empty()) {
+      snapshot_blob_path = per_process::cli_options->snapshot_blob;
+    } else {
+      snapshot_blob_path = std::string("snapshot.blob");
+      char buf[PATH_MAX_BYTES];
+      size_t cwd_size = sizeof(buf);
+      if (uv_cwd(buf, &cwd_size)) {
+        snapshot_blob_path =
+            std::string(buf) + kPathSeparator + std::string("snapshot.blob");
+      }
+    }
+
+    FILE* fp = fopen(snapshot_blob_path.c_str(), "w");
+    if (fp != nullptr) {
+      data.ToBlob(fp);
+      fclose(fp);
+    } else {
+      fprintf(stderr, "Cannot open %s", snapshot_blob_path.c_str());
+      result.exit_code = 1;
+    }
+  } else {
+    SnapshotData snapshot_data;
     Isolate::CreateParams params;
-    const std::vector<size_t>* indexes = nullptr;
+    const std::vector<size_t>* indices = nullptr;
```

Comment:
Not going to lie, I’m a big fan :)

Reply:
may I suggest `indexises`? No? Ok fine. lol 😝